src/ui/drawing/gdi/ImageRepository.cpp

```diff
@@ -413,11 +414,10 @@ HBITMAP ImageRepository::GetHBitmap(const ImageReference& pImage)
             if (hBitmap == nullptr)
                 return pImageRepository->GetDefaultImage(pImage.Type());
 
-            auto& pMutableImage = const_cast<ImageReference&>(pImage);
-            pMutableImage.SetData(reinterpret_cast<unsigned long>(hBitmap));
+            pImage.SetData(reinterpret_cast<unsigned long>(hBitmap));
 
             // ImageReference will release the reference
-            pImageRepository->AddReference(pMutableImage);
+            pImageRepository->AddReference(pImage);
```

Comment:
`AddReference` doesn't need `pImage` to be mutable. Making it accept a `const` should allow you to leave this method accepting a `const`. 

My goal state is to be able to pass around a `const ImageReference` and have it render an image to a surface. The details behind the implementation should not be important. In this case, we're storing the `HBITMAP` in a field in the `ImageReference` to avoid doing the resolution every time the image is drawn. The `ImageReference` itself is still considered immutable; we just don't have anywhere else to cache the `HBITMAP`. Since `HBITMAP` is GDI specific, all explicit references to `HBITMAP` must be in the GDI classes. 

If you don't want to make `SetData` const so you can call it from this method, just remove `GetData` and `SetData`. Now that we're a friend class, we can access `m_nData` directly.

Reply:
Made AddReference accept a const.